main.go

```diff
@@ -46,8 +48,34 @@ import (
 	"github.com/prometheus/pushgateway/storage"
 )
 
+var (
+	httpCnt = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "pushgateway_http_requests_total",
+			Help: "Total HTTP requests processed by the Pushgateway, excluding scrapes.",
+		},
+		[]string{"handler", "code", "method"},
+	)
+)
+
 func init() {
 	prometheus.MustRegister(version.NewCollector("pushgateway"))
+	prometheus.MustRegister(httpCnt)
+}
+
+func getFunctionName(i interface{}) string {
```

Comment:
Don't Know this should be used or not but it works fine if implemented. Made httpCnt common for all routes as used directly by route instrumentation. @beorn7 Any thoughts